Revise RDF data models

arnikz opened the pull request:
Updated input data and RDF models (TYPE annotations only) as documented [here](https://github.com/LINNAE-project/SFB-Annotator/tree/iss81/doc).

Commit:
Update data model: example 5_1 #81.

Commit:
Update rdf4j API to v3.5.0 #81.
- defines isIRI() method

Commit:
Update data model: example 5_1 #81.
- replace nhc:measuresOrDescribes with dwc:measurementType

Commit:
Update data model: example 5_1 #81.
- remove triple incl. rdf:type

Commit:
Update README #81.

Commit:
Update instance URL in example_6_1.json #81.

Commit:
Update data model: example 6_1 #81.

Commit:
Add obo prefix #22.

Commit:
Fix README.

Commit:
Update SVG plots #81.

Commit:
Update data model: example [4-6]_1 #81.
- fix: annotation dsw:derivedFrom image

Commit:
Update README.

Commit:
Fix verbatim text in example 7_1.json #81.

Commit:
Update data model: example 7_1 #81.
- add dwc:Event (class) and dwc:verbatimEventDate (property)

Commit:
Update README.

Commit:
Update data model: example 3_1 #81.
- add dwc:Location (class), dwc:verbatimLocality and dwciri:inDescribedPlace (properties)

Commit:
Update README.

Commit:
Update data model: example 2_1 #81.
- add dsw:derivedFromand dcterms:identifier (properties)

Commit:
Update data models: example [3,5,6]_1 #81.
- add dcterms:identifier (property)
- replace dwc->dwciri:measurementType (property)

Commit:
Update README.

Commit:
Update data model: example 1_1 #81.
- add dsw:derivedFrom & dwc:scientificName properties
- input JSON name/value pairs: belongstotaxon & rank are not used

Commit:
Update README.

Commit:
Update data model: example 2_1 #81.
- add foaf:name to foaf:Person instance

Commit:
Update JSON: example [1,5]_1 #81.
- use http://purl.obolibrary.org/obo/ to resolve NCBITaxon & NCIT IDs

Commit:
Update data models: example [5,6]_1 #81.
- change ns prefix: ncit->obo
- improve error handling for (instance) URLs

Commit:
Fix: handling language tags #55.

Commit:
Update README.

Commit:
Update example_5_1.json #81.
- use PATO instead of NCIT to refer to color

Commit:
Update data models: example *_1 #81.
- fix links with dsw:derivedFrom predicate

Commit:
Fix example_1_1.json #81.
- NCBITaxon->GBIF ID
- TAXORANK order->species

Commit:
Update ns prefixes.

Commit:
Update data model: example 1_1 #81.
- add dwc:Identification (class) and dwciri:toTaxon (property)
- remove unused vars

Commit:
Update data models: example *_* #81.
- fix ns prefixes: annot & gbif

Commit:
Add new plots of RDF graphs #22.
- replace rdf2svg.py with raptor2-utils (rapper) & graphviz (dot)

Commit:
Update CI #22.

Commit:
Update README #81.

Commit:
Fix date/time exception handling #74.

lisestork commented:
The visualisations look really great. Below some comments:

**1_1:** 
- Only evidence (in the form of a token, which is a) an image of an observation, and b) subimage of a measurement or fact such as a table or an anatomical entity) can be used with 'derivedFrom'. So dsw:derivedFrom should be removed here.
- Identification should not be a taxon, because then it would be a taxon class, the identification refers to the process of identification, so should receive a minted URI. See example 22 in https://dwc.tdwg.org/rdf/. In my opinion, we could mint a for the identification (or have a blank node for now) and link the triple gbif:9180402 dwc:taxonRank obo:TAXRANK_0000006 to the body of the annotation with the link owl:sameAs, indicating that we believe this taxon that is annotated is in fact the same as the one that is described by gbif. I believe it would be very important to trace the provenance of such statements, possibly using nanopublications. But that would be a next step  
- I am doubtful whether something can be both a class (e.g. dwc:Taxon) and a textualbody, although the graph seems to make a lot of sense like this. 

**2_1:** 
- same as last point 1_1 (semantic class vs textual body)

**3_1:**
- same as last point 1_1 (semantic class vs textual body)

**4_1:**
- same as last point 1_1 (semantic class vs textual body)
- Here, dsw:derivedFrom is used well, as dwc:Measurement is seen as a token. So, additionally, this node could have dsw:Token as its class. 

**5_1:** 
- Here, what we annotate should be (in my opinion) of type measurementType. So, it would not be true that the body is a measurement. However, I think that you are right in modelling it like this, we should possibly prefer pragmatism over being ontologically correct. Also, no class exists for MeasurementType. If we would like, however, we could create a class MeasurementType, and say: 
bodyIRI -> rdf:type -> MeasurementType. 
measurementOrfactIRI -> dwc:measurementType -> body1. 
measurementOrfactIRI -> rdf:type -> dwc:MeasurementOrFact. 
Let's discuss this further. 

**6_1:** 
- same as 5_1

**7_1:** 
- I think here we should say that the body is a date, and link the conceptual event to the date with either verbatimEvent or dwc:eventDate

Commit:
Update data model: example 7_1 #81.
- add ISO-formatted date field in addition to verbatim #74

Commit:
Update data models: example [4-6]_1 #81.
- add dsw:Token to Bnode

Commit:
Update data models: example [5-6]_1 #81.
- fix triple with dwciri:measurementType predicate

Commit:
Update data model: example 7_1 #81.
- fix triples with dwc:eventDate & dwc:verbatimEventDate predicates

Commit:
Update data model: example 1_1 #81.
- add owl:sameAs (property) and Bnode as dwc:Identification instance

Commit:
Remove old SVG plots.

Commit:
Update data model: example 1_1 #81.
- remove dsw:derivedFrom predicate

Commit:
Update README.md

Commit:
Use constant for OWL namespace #81.

Commit:
Fix: delete annotation triples #71.

Commit:
Update query to delete annotation #71.
- remove unused imports
- add exceptions
- add XSD namespace
- use both source img and selector

lisestork commented on data/rdf/local/example_1_1.ttl:
```diff
@@ -2,48 +2,52 @@
-taxon:9397 a dwc:Taxon;
-  dwc:taxonRank obo:TAXRANK_0000003 .
+gbif:9180402 dwc:taxonRank obo:TAXRANK_0000006;
+  owl:sameAs _:node1evgqka3hx9 .
```
As `gbif:9180402` `owl:sameAs` `_:node1evgqka3hx9` is a reification and not a fact, could we add this as a reification statement? Would be an interesting experiment. The modelling seems a bit more straightforward in RDF*, but still this seems like a good way to keep track of provenance of statements. 

ex:statementIRI a rdf:Statement;
      dcterms:creator orcid:0000-0002-2146-4803;
      rdf:subject gbif:9180402;
      rdf:predicate owl:sameAs;
      rdf:object _:node1evgqka3hx9 .

arnikz replied:
See #85.

lisestork reviewed (commented):
Everything looks very good. I just had one comment which we could discuss, otherwise I will approve the pull request.

Commit:
Update data model: example 1_1 #81.
- remove/add predicates: owl:sameAs/dcterms:identifier

arnikz commented:
@lisestork: I've updated the [example 1_1](https://github.com/LINNAE-project/SFB-Annotator/blob/iss81/doc/models/example_1_1.svg) as discussed, however, the following triples are currently no so useful IMO:smile:

https://github.com/LINNAE-project/SFB-Annotator/blob/5d2a07442d5b0d8f2cdb21a16d0087e16efe8ebc/data/rdf/local/example_1_1.ttl#L51-L52

Perhaps these triples could be added later when connecting two or more TYPE annotations. What do you think?

Commit:
Some example I/O files deprecated #81.
- example_1_{2,3,4}.{json,ttl}

lisestork commented:
> @lisestork: I've updated the [example 1_1](https://github.com/LINNAE-project/SFB-Annotator/blob/iss81/doc/models/example_1_1.svg) as discussed, however, the following triples are currently no so useful IMO😄
> 
> https://github.com/LINNAE-project/SFB-Annotator/blob/5d2a07442d5b0d8f2cdb21a16d0087e16efe8ebc/data/rdf/local/example_1_1.ttl#L51-L52
> 
> Perhaps these triples could be added later when connecting two or more TYPE annotations. What do you think?

I agree, they could be added later when connecting TYPE annotations.

Commit:
Update data model: example 1_1 #81.
- remove triples with dwc:Identification
